x-pack/plugins/apm/public/components/shared/Links/apm/AnomalyDetectionSetupLink.tsx

```diff
@@ -6,43 +6,73 @@
 import React from 'react';
 import { EuiButtonEmpty, EuiToolTip, EuiIcon } from '@elastic/eui';
 import { i18n } from '@kbn/i18n';
-import { ErrorCode } from '../../../../../common/anomaly_detection';
+import { useApmPluginContext } from '../../../../hooks/useApmPluginContext';
 import { APIReturnType } from '../../../../services/rest/createCallApmApi';
 import { APMLink } from './APMLink';
 import { getEnvironmentLabel } from '../../../../../common/environment_filter_values';
 import { useUrlParams } from '../../../../hooks/useUrlParams';
 import { useFetcher, FETCH_STATUS } from '../../../../hooks/useFetcher';
+import { useLicense } from '../../../../hooks/useLicense';
 
 export type AnomalyDetectionApiResponse = APIReturnType<
   '/api/apm/settings/anomaly-detection',
   'GET'
 >;
 
-const DEFAULT_DATA = { jobs: [], hasLegacyJobs: false, errorCode: undefined };
+const DEFAULT_DATA = { jobs: [], hasLegacyJobs: false };
 
 export function AnomalyDetectionSetupLink() {
   const { uiFilters } = useUrlParams();
   const environment = uiFilters.environment;
+  const plugin = useApmPluginContext();
+  const canGetJobs = !!plugin.core.application.capabilities.ml.canGetJobs;
+  const license = useLicense();
+  const hasValidLicense = license?.isActive && license?.hasAtLeast('platinum');
 
+  return (
+    <APMLink path="/settings/anomaly-detection">
+      <EuiButtonEmpty size="s" color="primary" iconType="inspect">
+        {ANOMALY_DETECTION_LINK_LABEL}
+      </EuiButtonEmpty>
+
+      {canGetJobs && hasValidLicense ? (
+        <MissingJobsAlert environment={environment} />
```

Comment:
Only retrieve ML data if the user has access to read ml jobs and the license is correct